src/app/cli/src/coda_archive_node_test.ml

```diff
@@ -0,0 +1,60 @@
+open Core
+open Async
+open Signature_lib
+open Coda_base
+
+let name = "coda-archive-node-test"
+
+let main () =
+  let logger = Logger.create () in
+  let largest_account_keypair =
+    Genesis_ledger.largest_account_keypair_exn ()
+  in
+  let largest_account_public_key =
+    Public_key.compress largest_account_keypair.public_key
+  in
+  let n = 2 in
+  let proposers i = if i = 0 then Some i else None in
+  let snark_work_public_keys i =
+    if i = 0 then Some largest_account_public_key else None
+  in
+  let is_archive_node i = i = 1 in
+  let%bind testnet =
+    Coda_worker_testnet.test logger n proposers snark_work_public_keys
+      Cli_lib.Arg_type.Sequence ~max_concurrent_connections:None
+      ~is_archive_node
+  in
+  let%bind new_block_pipe =
+    let%map pipe =
+      Coda_worker_testnet.Api.new_block testnet 1 largest_account_public_key
+    in
+    (Option.value_exn pipe).pipe
+  in
+  let num_blocks_to_wait = 5 in
+  let%bind _, observed_hashes =
+    Pipe.fold new_block_pipe ~init:(0, [])
+      ~f:(fun (i, acc_hashes) With_hash.{hash; _} ->
+        if i >= num_blocks_to_wait then Pipe.close_read new_block_pipe ;
+        Deferred.return (i + 1, hash :: acc_hashes) )
+  in
+  let observed_hashset = State_hash.Hash_set.of_list observed_hashes in
+  let%bind stored_transitions =
+    Coda_worker_testnet.Api.get_all_transitions testnet 1
+      largest_account_public_key
+  in
+  let stored_state_hashes =
+    State_hash.Hash_set.of_list
+      (List.map (Option.value_exn stored_transitions)
+         ~f:(fun {With_hash.hash; _} -> hash))
+  in
+  assert (
+    Hash_set.equal observed_hashset
```

Comment:
This doesn't seem like a strong enough statement. We want the sets to be identical, right? We don't want there to be stored state hashes that we didn't observe...